management/errors.go

```diff
@@ -0,0 +1,26 @@
+package management
+
+import (
+	"encoding/xml"
+	"fmt"
+)
+
+// AzureError represents an error returned by the management API. It has an error
+// code (for example, ResourceNotFound) and a descriptive message.
+type AzureError struct {
+	XMLName xml.Name `xml:"Error"`
+	Code    string
+	Message string
+}
+
+//Error implements the error interface for the AzureError type.
+func (e *AzureError) Error() string {
```

Comment:
Hhmm I don't know why this was initially defined on a pointer to comply to the `error` interface historically. This sounds like a bad idea however.

While we're at it can you please define this on `AzureError` (without pointer). Once you do that I guess you'll need to change one or two places in the code plus the test cases below (to remove `&`) but we will end up in a nicer world IMO. :hatching_chick: